demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,508 @@
+            emptyInferRequests.push(IRInfo(inferRequests[i], std::vector<IRInfo::IRImage>()));
+        }
+
+        auto startTime = std::chrono::system_clock::now();
+        auto currentTime = std::chrono::system_clock::now();
+        std::chrono::duration<double> elapsedSeconds = currentTime - startTime;
+        // ---------------------------------------------------------------------------------------------------
+        
+
+
+        // ===================================================================================================
+        do {
+            if (irCallbackException) std::rethrow_exception(irCallbackException);
+
+            if (isTestMode && elapsedSeconds.count() >= 10) {
+                isTestMode = false;
+
+                gridMat = GridMat(inputImagesCount, presenter, cv::Size(width, height), cv::Size(16, 9), avgFPS);
+                cv::Size gridMatSize = gridMat.getSize();
+                delay = ((FLAGS_delay == -1)
+                         ? static_cast<int>(avgFPS / (gridMatSize.width * gridMatSize.height) * 1000)
+                         : FLAGS_delay);
+            }
+
+            std::unique_lock<std::mutex> lock(mutex);
+        
+            if (!completedInferRequests.empty()) {
+                auto completedIRInfo = completedInferRequests.front();
+                completedInferRequests.pop();
+                mutex.unlock();
+                emptyInferRequests.push(IRInfo(completedIRInfo.ir, std::vector<IRInfo::IRImage>()));
+
+                // -----------------------------------Process output blobs------------------------------------
+                std::vector<unsigned> rightClasses = {};
+                for (size_t i = 0; i < completedIRInfo.images.size(); i++) {
+                    rightClasses.push_back(completedIRInfo.images[i].rightClass);
+                }
+
+                ClassificationResult res(completedIRInfo.ir.GetBlob(outputName), FLAGS_b, FLAGS_nt);
+                std::vector<unsigned> results = res.topResults(FLAGS_nt, *completedIRInfo.ir.GetBlob(outputName));
+                std::vector<std::string> predictedLabels = {};
+                for (size_t i = 0; i < FLAGS_b; i++) {
+                    bool isPredictionRight = false;
+                    std::string firstPredicted = labels[results[FLAGS_nt*i]];
+                    std::string predictedRight;
+                    for (size_t j = 0; j < FLAGS_nt; j++) {
+                        unsigned predictedClass = results[FLAGS_nt*i+j];
+                        if (predictedClass == rightClasses[i]) {
+                            isPredictionRight = true;
+                            predictedRight = labels[predictedClass];
+                            break;
+                        }
+                    }
+                    if (isPredictionRight) {
+                        predictedLabels.push_back(predictedRight);
+                        rightPredictionsCount++;
```

Comment:
These lines can be moved under `if (predictedClass == rightClasses[i])`. It will eliminate a need in `predictedRight` var.